### PR TITLE
Allow user to select type of animation, and polish the bounce animation

### PR DESCRIPTION
**Pull request by ansonyao**
Hi, thanks for the great lib, it has been very helpful to me. I am mainly using the bounce animation in the library as our designer wants it to be always bouncing regardless of the length of text. So I tweak the library to allow this. I also polished bounce animation. In current version, the delay is applied when it bounced right, but not applied when it bounced left, which makes it asymmetric if marqueeDelay is not zero. I also added a speed props such that user can select which speed they wants.

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -13,6 +13,12 @@ import PropTypes from 'prop-types'
 
 const { UIManager } = NativeModules
 
+export const TextTickAnimationType = Object.freeze({
+  auto: 'auto',
+  scroll: 'scroll',
+  bounce: 'bounce'
+})
+
```
Comment (deanhet) [inline index.js]: Can you add these options to the table of props in the README?

File: index.js [hunk 2]
```diff
@@ -114,39 +124,48 @@ export default class TextMarquee extends PureComponent {
...
           toValue:         10,
-          duration:        duration || children.length * 50,
+          duration:        duration || (this.distance) * scrollingSpeed,
           easing:          easing,
           isInteraction:   isInteraction,
           useNativeDriver: useNativeDriver
         })
       ]).start(({finished}) => {
+        if (finished) {
+          this._hasFinishedFirstLoop = true
```
Comment (deanhet) [inline index.js]: Just to follow previous convention and what's already there, can you remove `_` from method names?

File: index.js [hunk 3]
```diff
@@ -30,8 +36,10 @@ export default class TextMarquee extends PureComponent {
       PropTypes.string,
       PropTypes.array
     ]),
-    repeatSpacer:    PropTypes.number,
-    easing:          PropTypes.func
+    repeatSpacer:      PropTypes.number,
+    easing:            PropTypes.func,
+    animationType:     PropTypes.string, //(values should be from AnimationType, 'auto', 'scroll', 'bounce')
+    scrollingSpeed:    PropTypes.number //Will be ignored if you set duration directly.
```
Comment (deanhet) [inline index.js]: Additionally, this prop in the README too.

File: index.js (hunk 1 above)
Comment (ansonyao) [reply index.js]: Sure, I will update it on this weekend

Comment (ansonyao) [reply index.js]: Have updated the readme

File: index.js (hunk 2 above)
Comment (ansonyao) [reply index.js]: Have updated this

**Comment by deanhet** [comment]
Thanks for your work on this, 0.1.5 has just been released with these changes.

**Pull request merged**